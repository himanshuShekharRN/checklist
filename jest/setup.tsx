jest.mock('react-native-gesture-handler');
jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);
jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

// Mock the navigation object and its functions
export const mockNavigate = jest.fn();
export const mockOnPressButton = jest.fn();
export const useNavigation = () => ({
  navigate: mockNavigate,
  dispatch: jest.fn(),
  goBack: jest.fn(),
  navigation: {
    goBack: jest.fn(),
  },
});
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    createStackNavigator: jest.fn(),
    useNavigation: useNavigation,
    useRoute: () => ({
      params: {},
    }),
  };
});
