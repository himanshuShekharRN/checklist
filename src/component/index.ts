import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
