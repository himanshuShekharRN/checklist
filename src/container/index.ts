import {NationalChecklist} from './nationalChecklist';
import {DepartureDocCard} from './departureDocCard';
import {CustomTab} from './customTab';
import {MyCheckList} from './myChecklist';
import {PreDepartureList} from './preDepartureList';

export {
  NationalChecklist,
  DepartureDocCard,
  CustomTab,
  MyCheckList,
  PreDepartureList,
};
