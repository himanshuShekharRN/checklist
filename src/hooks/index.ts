import {usePreDepartureListData} from './usePreDepartureListData';
import {useChecklistCompletionStatus} from './useChecklistCompletionStatus';

export {useChecklistCompletionStatus, usePreDepartureListData};
