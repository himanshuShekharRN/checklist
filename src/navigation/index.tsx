import {NavigationContainer} from '@react-navigation/native';
import React from 'react';
import {CheckListStackNavigator} from './checkListStackNavigator';

export const Navigation = () => {
  return (
    <NavigationContainer>
      <CheckListStackNavigator />
    </NavigationContainer>
  );
};
