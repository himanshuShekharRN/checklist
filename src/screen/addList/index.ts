import {AddList} from './AddList.screen';

export {AddList};
