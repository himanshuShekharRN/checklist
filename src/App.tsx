import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
