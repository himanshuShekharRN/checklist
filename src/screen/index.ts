import {Checklist} from './checklist';
import {Home} from './home';
import {AddList} from './addList';
import {ViewList} from './viewList';

export {Checklist, Home, AddList, ViewList};
