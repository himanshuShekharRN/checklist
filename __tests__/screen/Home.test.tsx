import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';

import {Home} from '../../src/screen';
import {NavigationContainer} from '@react-navigation/native';
import {mockNavigate} from '../../jest/setup';

describe('Home Screen', () => {
  const tree = render(
    <NavigationContainer>
      <Home />
    </NavigationContainer>,
  );

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('snapshot created successfully', () => {
    render(
      <NavigationContainer>
        <Home />
      </NavigationContainer>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('onButton Press user navigates to Checklist screen', () => {
    render(
      <NavigationContainer>
        <Home />
      </NavigationContainer>,
    );
    const btnComponent = screen.getByTestId('homeBtn');
    fireEvent.press(btnComponent);

    expect(mockNavigate).toHaveBeenCalledWith('CheckList');
    expect(mockNavigate).not.toHaveBeenCalledWith('ViewList');
  });
});
