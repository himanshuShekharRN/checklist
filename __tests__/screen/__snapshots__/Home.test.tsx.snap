// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen snapshot created successfully 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="homeBtn"
  >
    <View
      testID="customIcon"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#005AA5",
              "fontSize": 380,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        îŒ•
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
